python/LLM Examples/PrivateAI_QA_finetune.ipynb

```diff
@@ -0,0 +1,8182 @@
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "20dd8ef5-3d55-4374-8c79-a619a2fcd548",
+      "metadata": {
+        "tags": [],
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "20dd8ef5-3d55-4374-8c79-a619a2fcd548",
+        "outputId": "dce094e9-e6c3-4cd8-da33-69e3e3be5e94"
+      },
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            ">> User:Does CUDA support ISO\n",
+            "DialoGPT: Yes, CUDA support in ISO is supported.\n",
+            ">> User:who made an AI system created the sings Christmas songs?\n",
+            "DialoGPT: The researchers from the University of California, Berkeley developed an AI system that automatically sings Christmas songs to humans.\n"
+          ]
+        },
+        {
+          "output_type": "stream",
+          "name": "stderr",
+          "text": [
+            "KeyboardInterrupt: Interrupted by user\n"
+          ]
+        }
+      ],
+      "source": [
+        "\n",
+        "## Sample Questions to test\n",
+        "# Does CUDA support ISO\n",
+        "#  name an organization with deep AI research\n",
+        "# who made an AI system created the sings Christmas songs?\n",
+        "\n",
+        "start_chat(OUTPUT_DESTINATION  + 'fine_tuned_DialoGPT')"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "fc9f177a-fd8b-48a3-8bde-2a10022e2fdc",
+      "metadata": {
+        "id": "fc9f177a-fd8b-48a3-8bde-2a10022e2fdc"
+      },
+      "source": [
+        "# AI-Powered Redaction of Domain-Specific Data\n",
+        "\n",
+        "While our previous efforts focused on fine-tuning the language model using un-redacted data, this fined tuned model risks leaking PII data. To address this concern, we're employing AI-powered redaction on our domain-specific dataset, courtesy of the Private AI open API, before the fine-tuning process. This method ensures sensitive information within the dataset is anonymized or removed, thereby protecting individual privacy and confidential information.\n",
+        "\n",
+        "__Important__\n",
+        "To access the cloud API requires a key. Please visit https://www.private-ai.com/start and click on \"Get Started Now\" to get a free key."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "2c9b4370-9651-4d90-846e-be51fd76f00f",
+      "metadata": {
+        "tags": [],
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "2c9b4370-9651-4d90-846e-be51fd76f00f",
+        "outputId": "1e87dbd6-afac-4067-8696-fb7983365746"
+      },
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "CPU times: user 3 µs, sys: 0 ns, total: 3 µs\n",
+            "Wall time: 5.96 µs\n"
+          ]
+        }
+      ],
+      "source": [
```

Comment:
This step gave me a 429 error, too many requests. We limit our API keys to 75 calls per day, and 10 calls per minute, so perhaps this should be updated.
![Screenshot 2024-02-27 at 11 50 16 AM](https://github.com/privateai/deid-examples/assets/30555712/e3b7696e-8c39-44f1-ad54-a7729df74d08)

Reply:
I can't really continue with this notebook from here, unfortunately.

Reply:
That's changed since this was first launched. I'll update the batch size to use fewer API calls.